src/Storages/StorageFuzzJSON.cpp

```diff
@@ -0,0 +1,511 @@
+        , config(config_)
+        , rnd(config.random_seed)
+        , root(root_)
+    {
+    }
+    String getName() const override { return "FuzzJSON"; }
+
+protected:
+    Chunk generate() override
+    {
+        Columns columns;
+        columns.reserve(block_header.columns());
+        columns.emplace_back(createColumn());
+
+        return {std::move(columns), block_size};
+    }
+
+private:
+    ColumnPtr createColumn();
+
+    UInt64 block_size;
+    Block block_header;
+
+    StorageFuzzJSON::Config config;
+    pcg64 rnd;
+
+    std::shared_ptr<Node> root;
+};
+
+ColumnPtr FuzzJSONSource::createColumn()
+{
+    auto column = ColumnString::create();
+    ColumnString::Chars & data_to = column->getChars();
+    ColumnString::Offsets & offsets_to = column->getOffsets();
+
+    offsets_to.resize(block_size);
+    IColumn::Offset offset = 0;
+
+    for (size_t row_num = 0; row_num < block_size; ++row_num)
+    {
+        WriteBufferFromOwnString out;
+        fuzzJSONObject(root, out, config, rnd, 0);
+        auto data = out.str();
+
+        size_t data_len = data.size();
+
+        IColumn::Offset next_offset = offset + data_len + 1;
+        data_to.resize(next_offset);
+
+        std::copy(data.begin(), data.end(), &data_to[offset]);
+
+        data_to[offset + data_len] = 0;
+        offsets_to[row_num] = next_offset;
+
+        offset = next_offset;
+    }
+
+    return column;
+}
+
+}
+
+StorageFuzzJSON::StorageFuzzJSON(
+    const StorageID & table_id_,
+    const ColumnsDescription & columns_,
+    const String & comment_,
+    const Config & config_,
+    const String & json_str_)
+    : IStorage(table_id_), config(config_), json_root(parseJSON(json_str_))
+{
+    StorageInMemoryMetadata storage_metadata;
+    storage_metadata.setColumns(columns_);
+    storage_metadata.setComment(comment_);
+    setInMemoryMetadata(storage_metadata);
+}
+
+Pipe StorageFuzzJSON::read(
+    const Names & column_names,
+    const StorageSnapshotPtr & storage_snapshot,
+    SelectQueryInfo & /*query_info*/,
+    ContextPtr /*context*/,
+    QueryProcessingStage::Enum /*processed_stage*/,
+    size_t max_block_size,
+    size_t num_streams)
+{
+    storage_snapshot->check(column_names);
+
+    Pipes pipes;
+    pipes.reserve(num_streams);
+
+    const ColumnsDescription & our_columns = storage_snapshot->metadata->getColumns();
+    Block block_header;
+    for (const auto & name : column_names)
+    {
+        const auto & name_type = our_columns.get(name);
+        MutableColumnPtr column = name_type.type->createColumn();
+        block_header.insert({std::move(column), name_type.type, name_type.name});
+    }
+
+    for (UInt64 i = 0; i < num_streams; ++i)
+        pipes.emplace_back(std::make_shared<FuzzJSONSource>(max_block_size, block_header, config, json_root));
+
+    return Pipe::unitePipes(std::move(pipes));
+}
+
+void registerStorageFuzzJSON(StorageFactory & factory)
+{
+    factory.registerStorage(
```

Comment:
Can we also add test cases fro `FuzzJSON` storage engine?

Is this query supported?

```sql
CREATE TABLE test_table (value String) ENGINE = FuzzJSON;
```